server/routes/referrals/referralFormPresenter.ts

```diff
@@ -208,7 +208,7 @@ class FormSectionBuilder {
                       `service-category/${serviceCat.id}/desired-outcomes`,
                       index === 0
                         ? this.taskValues.relevantSentence
-                        : this.taskValues.complexityLevel(this.intervention.serviceCategories[index - 1].id)
+                        : this.taskValues.complexityLevel(this.referral.serviceCategoryIds![index - 1])
```

Comment:
Although it works, perhaps we should also do a sort of this.referral.serviceCategoryIds?

Reply:
I'm still a bit confused about the sorting logic 😕 would you mind explaining what that would do for us?

Reply:
it's only because the interventions is sorted based on the id.

the service category ids should already be sorted by the backend - but it provides additional protection if the contract with the back-end changes

Reply:
Have pushed this change to sort them.